### ISSUE TITLE
DB/Quest: Bring Em Back Alive dismounts players in intended quest area

### ISSUE THREAD

**Issue by Foereaper** [TrinityCore/TrinityCore#20078]
**Description:**

Attached SQL fixes the issues related to this quest, see current behaviour.

**Current behaviour:**

Currently the Kodo vehicle dismounts the player even though the player is within the intended questing area.

**Expected behaviour:**

The Kodo should allow the players to be in the specific sub-zones around the questing area and not throw the player off.

**Branch(es):** both

**TC rev. hash/commit:** 21b03e7

**TDB version:**  335.63

**Operating system:** Windows

**Suggested fix:**

```sql
-- Fixes "Bring 'Em Back Alive" Borean quest
DELETE FROM `conditions` WHERE `SourceTypeOrReferenceId`=16 AND `SourceEntry`=25596 AND `ConditionTypeOrReference`=23 AND `ElseGroup` IN (0,1,2,3,4);
INSERT INTO `conditions` (`SourceTypeOrReferenceId`,`SourceGroup`,`SourceEntry`,`SourceId`,`ElseGroup`,`ConditionTypeOrReference`,`ConditionTarget`,`ConditionValue1`,`ConditionValue2`,`ConditionValue3`,`NegativeCondition`,`ErrorType`,`ErrorTextId`,`ScriptName`,`Comment`) VALUES
(16, 0, 25596, 0, 0, 23, 0, 3537, 0, 0, 0, 0, 0, '','Bring Em Back Alive: Dismount when player outside of intended zone: Borean Tundra'),
(16, 0, 25596, 0, 1, 23, 0, 4144, 0, 0, 0, 0, 0, '','Bring Em Back Alive: Dismount when player outside of intended zone: Warsong Farms Outpost'),
(16, 0, 25596, 0, 2, 23, 0, 4143, 0, 0, 0, 0, 0, '','Bring Em Back Alive: Dismount when player outside of intended zone: Warsong Slaughterhouse'),
(16, 0, 25596, 0, 3, 23, 0, 4142, 0, 0, 0, 0, 0, '','Bring Em Back Alive: Dismount when player outside of intended zone: Warsong Grannary'),
(16, 0, 25596, 0, 4, 23, 0, 4141, 0, 0, 0, 0, 0, '','Bring Em Back Alive: Dismount when player outside of intended zone: Torps Farm');
```

**Comment by tkrokli** [comment]
Thank you for posting again and in the right place,

I would also like to suggest a few improvements:
- Use ```sql (or ~~~sql ) on top of your query for SQL syntax highlighting (cosmetic, but practical for reading the code in your post).
- Always use `DELETE FROM` before any `INSERT INTO` commands are used in a query. This is needed to make new queries repeatable.
- Always put back ticks ( ` ) around table names and column names in query commands (some column names can be SQL keywords).

Example:
```sql
DELETE FROM `conditions` WHERE `SourceTypeOrReferenceId`=16 AND `SourceEntry`=25596 AND `ConditionTypeOrReference`=23 AND `ElseGroup` IN (1,2,3,4);
INSERT INTO `conditions` (`SourceTypeOrReferenceId`,`SourceGroup`,`SourceEntry`,`SourceId`,`ElseGroup`,`ConditionTypeOrReference`,`ConditionTarget`,`ConditionValue1`,`ConditionValue2`,`ConditionValue3`,`NegativeCondition`,`ErrorType`,`ErrorTextId`,`ScriptName`,`Comment`) VALUES
```

I know many devs don't care much about the details in the comments, but you may want to include the area names in the comments,

Zone | Name | MapID | AreaID
-- | -- | -- | --
4141 | Torp's Farm | 571 | 3537
4142 | Warsong Granary | 571 | 3537
4143 | Warsong Slaughterhouse | 571 | 3537
4144 | Warsong Farms Outpost | 571 | 3537

because TDB 335.63 already contains a condition `CONDITION_SOURCE_TYPE_CREATURE_TEMPLATE_VEHICLE` (16) for `creature_template.entry` 25596 (Infected Kodo Beast) and `CONDITION_AREAID` (23) for AreaID 3537 (Borean Tundra):
```sql
SELECT * FROM `conditions` WHERE `SourceTypeOrReferenceId`=16 AND `SourceEntry`=25596;
```
```
SourceTypeOrReferenceId  SourceGroup  SourceEntry  SourceId  ElseGroup  ConditionTypeOrReference  ConditionTarget  ConditionValue1  ConditionValue2  ConditionValue3  NegativeCondition  ErrorType  ErrorTextId  ScriptName  Comment                                  
-----------------------  -----------  -----------  --------  ---------  ------------------------  ---------------  ---------------  ---------------  ---------------  -----------------  ---------  -----------  ----------  -----------------------------------------
                     16            0        25596         0          0                        23                0             3537                0                0                  0          0            0              Dismount player when not in intended zone
```

**Comment by Foereaper** [comment]
Updated with suggestions :)

**Comment by tkrokli** [comment]
Just a little F.Y.I. : an alternative to escape apostrophes ( ' ) with either double apostrophes ( ' ' ) or backslash ( \' ) in comment text is to use double quotation marks ( " " ) before and after the text when it contains contractions and "possessive form" like "Torp's Farm" to avoid breaking the comment delimiters. Example:
```sql
(16, 0, 25596, 0, 4, 23, 0, 4141, 0, 0, 0, 0, 0, '',"Bring 'Em Back Alive: Dismount when player outside of intended zone: Torp's Farm");
```
(this will also make it easier to use quest/item/NPC names containing apostrophes, like this quest "Bring 'Em Back Alive")

Also: **unless** there is more wrong data than correct data in an existing DB row, use `UPDATE` queries to fix the few bad data entries instead of replacing the entire row.

Example:
```sql
UPDATE `conditions` SET `Comment`= 'Bring Em Back Alive: Dismount when player outside of intended zone: Borean Tundra'
 WHERE `SourceTypeOrReferenceId`=16 AND `SourceEntry`=25596 AND `ConditionTypeOrReference`=23 AND `ElseGroup`=0;
```

Lastly, if you will allow me to be nit-picky about the comment line on top, I would like to see more specific details:
```sql
--  Allow player to mount Infected Kodo Beast inside all the intended zones in Borean Tundra
```

**Comment by tkrokli** [comment]
Closed in 734fab01bb93c7b9e79e9f11e30a9782ddb31a77 - @TrinityCoreBot is sleeping?
